src/eos-updater-apply.c

```diff
@@ -120,42 +161,74 @@ apply (GTask *task,
     {
       g_autofree gchar *rev = NULL;
 
-      if (!ostree_repo_resolve_rev (repo, orig_refspec, TRUE, &rev, &error))
-        goto error;
+      if (!ostree_repo_resolve_rev (repo, orig_refspec, TRUE, &rev, error))
+        return FALSE;
 
       if (rev)
         {
-          if (!ostree_repo_prepare_transaction (repo, NULL, cancel, &error))
-            goto error;
+          if (!ostree_repo_prepare_transaction (repo, NULL, cancel, error))
+            return FALSE;
 
           ostree_repo_transaction_set_refspec (repo, orig_refspec, NULL);
 
-          if (!ostree_repo_commit_transaction (repo, NULL, cancel, &error))
-            goto error;
+          if (!ostree_repo_commit_transaction (repo, NULL, cancel, error))
+            return FALSE;
         }
     }
 
   if (!ostree_sysroot_simple_write_deployment (sysroot,
-                                               NULL,
+                                               osname,
                                                new_deployment,
-                                               merge_deployment,
+                                               booted_deployment,
                                                0,
                                                cancel,
-                                               &error))
-    goto error;
+                                               error))
+    return FALSE;
 
   newbootver = ostree_deployment_get_deployserial (new_deployment);
 
-  g_task_return_boolean (task, bootversion != newbootver);
-  goto cleanup;
+  if (!eos_extensions_save (data->extensions,
+                            repo,
+                            cancel,
+                            error))
+    return FALSE;
+  g_clear_object (&data->extensions);
+
+  head_commit_timestamp = g_date_time_new_from_unix_utc (ostree_commit_get_timestamp (update_commit));
+
+  if (!eos_avahi_generate_service_file (repo,
+                                        head_commit_timestamp,
+                                        error))
+    return FALSE;
```

Comment:
I think the `.path`-file based approach is probably the best way to go, with `PathChanged=/ostree/repo/refs/heads/ostree/1/1/0; PathChanged=/etc/eos-updater/eos-update-server.conf; PathChanged=/usr/share/eos-updater/etc/eos-update-server.conf`. It would run an `eos-updater-avahi` service to update the Avahi `.service` file.

`/etc/eos-updater/eos-update-server.conf` would be a key file containing a single option: whether to enable serving of updates over the LAN. We could then enable `eos-update-server.socket` by default. If the option is disabled, `eos-update-server` would bail as soon as it’s run; and the `eos-updater-avahi` would remove the Avahi `.service` file.

The problem I see with this approach is that `eos-updater-avahi.service` and `eos-update-server.socket` have to have the same enabled/disabled state. It’s no good if one of them is enabled and the other isn’t. We can’t trivially roll `eos-updater-avahi` into `eos-update-server`, because the latter may run for a very long time, and once it’s running, it’s running (so we can’t just update the Avahi `.service` file on startup). We’d end up reimplementing all the `.path` checks in code. I guess this can be solved by adding appropriate `Also=` and `Requires=` dependencies between the two units.